File: MS.NET/Language/OOP/interval.cs
//partial class - a class whose source is spread across different files
partial class Interval
{
	public int Minutes {get;}

	public int Seconds {get;}

	//partial method - a method of partial class which can be defined in another source file
	partial void OnInitialize();

	public Interval(int min, int sec)
	{
		Minutes = min + sec / 60;
		Seconds = sec % 60;
		OnInitialize();
	}

	public int Time => 60 * Minutes + Seconds;

	public override string ToString()
	{
		return $"{Minutes}:{Seconds:00}";
	}

	public override int GetHashCode() => Minutes + Seconds;

	public override bool Equals(object other)
	{
		if(other is Interval that)
		{
			return (this.Minutes == that.Minutes) && (this.Seconds == that.Seconds);
		}

		return false;
	}

}

File: Java/Fundamentals/Language/OOP/Interval.java
class Interval{

	private int min, sec;

	public Interval(int m, int s){
		min = m + s / 60;
		sec = s % 60;
	}

	public int minutes(){
		return min;
	}

	public int seconds(){
		return sec;
	}

	public int time(){
		return min + 60 * sec;
	}

	public String toString(){
		return String.format("%d:%2d", min, sec);
	}

	public int hashCode(){
		return min + sec;
	}

	public boolean equals(Object other){
		if(other instanceof Interval){
			Interval that = (Interval)other;
			return (this.min == that.min) && (this.sec == that.sec);
		}
		return false;
	}
}

